chia/wallet/wallet_state_manager.py

```diff
@@ -787,7 +787,7 @@ async def handle_did(
 
         launch_id: bytes32 = bytes32(bytes(singleton_struct.rest().first())[1:])
         if derivation_record is None:
-            if new_p2_puzhash == old_inner_puzhash or p2_puzzle == new_p2_puzhash:
+            if new_p2_puzhash != old_inner_puzhash and p2_puzzle != new_p2_puzhash:
```

Comment:
can you explain this change?

Reply:
When the wallet cannot find a key for the hint (new_p2_puzhash), it will check if it's because of the hint bug.
 In this case, if the hint doesn't equal the previous didDIDinner puzhash or p2_puzzle hash, we know the user may update the p2_address. We have to wait for an additional spend to confirm the p2_puzzle is changed then delete the wallet.

Reply:
thanks, lets add a comment here to explain the logic